R/rdeephaven/R/table_handle_wrapper.R

```diff
@@ -35,6 +35,24 @@ TableHandle <- R6Class("TableHandle",
                 you are trying to call the constructor of TableHandle directly, which is not advised.")
             }
             private$internal_table_handle <- table_handle
+            private$is_static_field <- private$internal_table_handle$is_static()
+        },
+
+        #' @description
+        #' Whether the table referenced by this TableHandle is static or not.
+        #' @return TRUE if the table is static, FALSE if the table is ticking.
+        is_static = function() {
+            return(private$is_static_field)
+        },
+
+        #' @description
+        #' Number of rows in the table referenced by this TableHandle. This value is only well-defined for static tables.
+        #' @return The number of rows in the table, if the table is static.
+        num_rows = function() {
+            if(!private$is_static_field) {
+                stop("The table referenced by this TableHandle is not a static table, so the number of rows is not well-defined.")
+            }
+            return(private$internal_table_handle$num_rows())
         },
```

Comment:
The documentation and what this is doing is not what we want.  Viable options are:
1) Rerequest the size when the method is called, and then return the size.
2) Raise an error that the size is not yet supported for dynamic tables.
The whole "not well defined" thing makes no sense.  For dynamic tables, the value may change, but it is well defined at the instant it is collected.
Eventually (1) needs to happen, but for the sake of shipping this, (2) is ok.  Just add a TODO.

Reply:
Also, is it more idiomatic for this to be `num_rows` or `size`.  I think `num_rows` is more clear.  What does dataframe do?

Reply:
data frame uses typical R styling of `nrow(dataframe)`. We can implement that later. Should this method be named `nrow` to be kept consistent with a future implementation that enables `nrow(TableHandle)`?

Reply:
If it is more idiomatc, I would do `nrow`... but it looks like `nrow` is a function that you would implement that would call whatever this function is called.

Reply:
It is more idiomatic with typical R, but I do not think it's what a user would expect with an R6 class where everything else is done with `$`. In any case, when `f` is implemented for an R6 class, the corresponding method that gets called is also called `f`.